### ISSUE TITLE
[Prefetching] Impossible to pre-fetch several presentations or to open a pre-fetched presentation several times

### ISSUE THREAD

**Issue by hatem-72** [Purchasely/Purchasely-ReactNative#59]
I struggle using the `Purchasely.fetchPresentation` method.
Here are two use cases causing issues:

- I can only present a pre-fetched `presentation` once.
```ts
const presentation = await Purchasely.fetchPresentation({
    placementId: 'some-placement'
});
// This one works ✅
await Purchasely.presentPresentation({
    presentation: presentation
});
// This one fails with error "io.purchasely error 2." ❌
await Purchasely.presentPresentation({
    presentation: presentation
});
```

- I can only prefetch one single presentation. If I do more than one the second call will freeze (no error, it just never resolves)
```ts
// First one resolves ✅
const presentation = await Purchasely.fetchPresentation({
    placementId: 'some-placement'
});
// Second one freezes 🥶
const presentation = await Purchasely.fetchPresentation({
    placementId: 'some-other-placement'
});
```

Am I using the API badly ? Is it a bug ? Is it specific to the react-native library ? (I am testing on IOS device)

**Comment by kherembourg** [comment]
Hello @hatem-72 
Thanks for your report, we are checking and I will get back to you soon

**Comment by kherembourg** [comment]
Hello @hatem-72 
It has been fixed with versions 2.7.4 and 4.0.1, sorry for the inconvenience

We suggest to always call `fetchPresentation()` before `presentPresentation()` but you can call it multiple times now if you want to open the same paywall. Be aware though that if you made any changes to user attributes or user status in general, it won't be updated until you call `fetchPresentation()` again
